azurecosmos/src/main/java/com/yahoo/ycsb/db/AzureCosmosClient.java

```diff
@@ -298,20 +284,41 @@ public Status scan(String table, String startkey, int recordcount, Set<String> f
 
   @Override
   public Status update(String table, String key, Map<String, ByteIterator> values) {
+    // Azure Cosmos does not have patch support.  Until then we need to read
+    // the document, update in place, and then write back.
+    // This could actually me made more efficient by using a stored procedure
+    // and doing the read/modify write on the server side.  Perhaps
+    // that will be a future improvement.
+
     String documentLink = getDocumentLink(this.databaseName, table, key);
-    Document document = getDocumentDefinition(key, values);
+    ResourceResponse<Document> updatedResource = null;
+    ResourceResponse<Document> readResouce = null;
+    RequestOptions reqOptions = null;
+    Document document = null;
+
+    try {
+      reqOptions = getRequestOptions(key);
```

Comment:
can we just embed the getRequestOptions() call into the readDocument() call? So the local variable reqOptions won't need to be introduced here.

Reply:
Its actually used a couple of times in this function so I avoid creating multiple objects.